### PR TITLE
run accumulated proofs through the main merkle tree

### PR DESCRIPTION
**Pull request by maxtaco**
- wasn't that bad, just have to squirrel away linkMaps when we got them

### CODE REVIEW

File: go/teams/proofs.go [hunk 1]
```diff
@@ -86,3 +90,58 @@ func (p *proofSetT) AddNeededHappensBeforeProof(a proofTerm, b proofTerm) *proof
...
+// check a single proof. Call to the merkle API enddpoint, and then ensure that the
+// data that comes back fits the proof and previously checked sighcain links.
+func (p proof) check(ctx context.Context, g *libkb.GlobalContext) error {
+	triple, err := p.lookupMerkleTreeChain(ctx, g)
+	if err != nil {
+		return err
+	}
+	laterSeqno := triple.Seqno
+	earlierSeqno := p.a.sigMeta.SigChainLocation.Seqno
+	if earlierSeqno > laterSeqno {
```
Comment (mlsteele) [inline go/teams/proofs.go]: I'm confused. Can't these seqnos be on different chains?

Comment (maxtaco) [inline go/teams/proofs.go]: no this is the sitewide merkle seqno.

Comment (mlsteele) [inline go/teams/proofs.go]: `p.a.sigMeta.SigChainLocation.Seqno` is a merkle seqno?
It's from https://github.com/keybase/client/blob/master/go/teams/chain_parse.go#L129 though

Comment (mlsteele) [inline go/teams/proofs.go]: resolved: they're both on chain `a`

File: go/teams/proofs.go [hunk 2]
```diff
@@ -86,3 +90,58 @@ func (p *proofSetT) AddNeededHappensBeforeProof(a proofTerm, b proofTerm) *proof
...
+	return ret, nil
+}
+
+// check a single proof. Call to the merkle API enddpoint, and then ensure that the
+// data that comes back fits the proof and previously checked sighcain links.
+func (p proof) check(ctx context.Context, g *libkb.GlobalContext) error {
+	triple, err := p.lookupMerkleTreeChain(ctx, g)
+	if err != nil {
+		return err
+	}
```
Comment (mlsteele) [inline go/teams/proofs.go]: as per our discussion, how about a comment
```
laterSeqno is the tail of chain A at the time when B was signed
earlierSeqno is the tail of chain A at the time when A was signed
```

File: go/teams/loader2.go [hunk 3]
```diff
@@ -144,32 +144,32 @@ func (l *TeamLoader) checkStubbed(ctx context.Context, arg load2ArgT, link *chai
...
 	defer l.G().CTrace(ctx, fmt.Sprintf("TeamLoader#loadUserForSigVerification(%d)", int(inner.Seqno)), func() error { return err })()
 	keySection := inner.Body.Key
 	if keySection == nil {
-		return nil, nil, libkb.NoUIDError{}
+		return nil, nil, nil, libkb.NoUIDError{}
 	}
 	uid := keySection.UID
 	kid := keySection.KID
-	user, key, err = l.G().GetUPAKLoader().LoadKeyV2(ctx, uid, kid)
+	user, key, linkMap, err = l.G().GetUPAKLoader().LoadKeyV2(ctx, uid, kid)
```
Comment (mlsteele) [inline go/teams/loader2.go]: This `linkMap` must be as up to date as the link we are checking. Does `LoadKeyV2` give us that, or could it be stale?

**Comment by maxtaco** [comment]
only need to be up-to-date enough to give us the KID we asked for, which it does give us

**Comment by mlsteele** [comment]
but isn't it the same link map used here? https://github.com/keybase/client/pull/7605/files#diff-b7515a197b2123085dab8910ed3535ddR125

**Comment by maxtaco** [comment]
you're right, good point

**Comment by maxtaco** [comment]
ok, all known points addressed

**Pull request merged**